msal/application.py

```diff
@@ -444,6 +444,8 @@ def __init__(
             New in version 1.19.0.
 
         :param boolean allow_broker:
+            This parameter is NOT applicable to :class:`ConfidentialClientApplication`.
```

Comment:
Maybe a positive statement?

> This parameter is ONLY applicable to `PublicClientApplication`.

Reply:
I believe the emphasis of this PR and its corresponding issue were to call out the only invalid situation among the three. So, taking your suggestion into account, we can change it to:

> This parameter is applicable to `PublicClientApplication` and its base class `ClientApplication`, but NOT applicable to `ConfidentialClientApplication`.

How about that?